src/CoffeeShop.java

```diff
@@ -0,0 +1,85 @@
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+public class CoffeeShop {
+    String name;
+    double x;
+    double y;
+    double distance;
+
+    public CoffeeShop(String name, double x, double y, double distance) {
+        this.name = name;
+        this.x = x;
+        this.y = y;
+        this.distance = distance;
+    }
+
+    // main method, expecting 3 command-line arguments (user x coords, user y coords, URL of the coffee shop CSV file)
+    public static void main(String[] args) {
+        if(args.length != 3){
+            System.out.println("Usage: <user_x_coordinate> <user_y_coordinate> <shop_data_url>");
+            return;
+        }
+        try{
+            // parse user coordinates
+            double x = Double.parseDouble(args[0]);
+            double y = Double.parseDouble(args[1]);
+            String shopUrl = args[2];
+
+            //fetch coffee shops and calculates the distance
+            List<CoffeeShop> coffeeShopList = fetchCoffeeShops(shopUrl, x, y);
+
+            // sort shops by distance
```

Comment:
The comments don't add any value here, ideally the code should be self explanatory, in rare cases comments should be added to explain the why not the how/what